### PR TITLE
fix to infer sourceDataType in cross store joins

### PR DESCRIPTION
**Pull request by theodosis1989**
#### What type of PR is this?

- Bug Fix

#### What does this PR do / why is it needed ?

In cross store joins the relational type is null for columns that are generated from ->extend.
This fix infers the relational type from the domain type.

### CODE REVIEW

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/relationalMappingExecution.pure [hunk 1]
```diff
@@ -42,6 +42,10 @@ import meta::relational::milestoning::*;
 import meta::pure::router::routing::*;
 import meta::pure::router::store::metamodel::*;
 
+Class meta::relational::mapping::DomainTypeToSourceDataTypePair extends Pair<String, meta::relational::metamodel::datatype::DataType>
```
Comment (ghub-real) [inline legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/relationalMappingExecution.pure:45]: we don't need this, right?

Comment (theodosis1989) [reply legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/relationalMappingExecution.pure:45]: it's used in the getSourceDataTypeFromType to create the Pairs. I followed the same pattern as in the pureToSqlQuery where it maps the function expression to a method

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/relationalMappingExecution.pure [hunk 2]
```diff
@@ -81,11 +81,13 @@ function meta::relational::mapping::generateInstantiationExecutionNode(sq:meta::
                                                                   offset = $paths->indexOf($p),
                                                                   sourceDataType = if($p.second.relationalType->isNotEmpty(),
                                                                                       | $p.second.relationalType,
-                                                                                      | $p.second.propertyMapping->match([
+                                                                                      | let sourceDataTypeFromMapping = $p.second.propertyMapping->match([
```
Comment (aormerod-gs) [inline legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/relationalMappingExecution.pure:84]: As discussed, can we handling this by changing the match statement to handle more precise multiplicity cases

**Review by aormerod-gs — APPROVED** [review]

**Review by gs-ssh16 — APPROVED** [review]

**Pull request merged**